### PR TITLE
Sharedlib

### PR DESCRIPTION
**Pull request by nDy**
Base sample calls in shared library

### CODE REVIEW

File: src/main/featuremine.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+typedef struct int_obj {
```
Comment (petver29) [inline src/main/featuremine.h]: I would suggest moving actual definition to c++ file. Leave just the declaration here.

File: src/main/featuremine.cpp [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#include "featuremine.h"
```
Comment (petver29) [inline src/main/featuremine.cpp]: Put extern "C" around include. We want to make sure that shared library is C ABI compatible. Python does not work well with C++.

Comment (nDy) [reply src/main/featuremine.cpp]: This was fixed in: https://github.com/nDy/FeatureMinePublic/pull/1/commits/d85b4db999e901fbf9c29c0c84459dce33137625

File: src/main/featuremine.h (hunk 1 above)
Comment (nDy) [reply src/main/featuremine.h]: This was fixed in: https://github.com/nDy/FeatureMinePublic/pull/1/commits/d85b4db999e901fbf9c29c0c84459dce33137625

**Pull request merged**